### PR TITLE
Issue #644 : dapr sidecar kubernetes services are not removed

### PR DESCRIPTION
**Pull request by shalabhms**
# Description

Uninstalling a service and pod  is not removing the dapr side car service.

## Issue reference

https://github.com/dapr/dapr/issues/644

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing

### CODE REVIEW

File: pkg/operator/handlers/dapr_handler.go [hunk 1]
```diff
@@ -134,4 +151,21 @@ func (h *DaprHandler) ObjectUpdated(old interface{}, new interface{}) {
...
 func (h *DaprHandler) ObjectDeleted(obj interface{}) {
+	h.deploymentsLock.Lock()
+	defer h.deploymentsLock.Unlock()
+
+	deployment := obj.(*appsv1.Deployment)
+	annotated := h.isAnnotatedForDapr(deployment)
+	if annotated {
+		id := h.getDaprID(deployment)
+		if id == "" {
+			log.Errorf("skipping service deletion: id for deployment %s is empty", deployment.GetName())
```
Comment (youngbupark) [inline pkg/operator/handlers/dapr_handler.go]: I would use warning log here instead of error.

File: pkg/kubernetes/kubernetes.go [hunk 2]
```diff
@@ -63,6 +63,18 @@ func CreateService(service *corev1.Service, namespace string) error {
...
+func DeleteService(serviceName string, namespace string) error {
+	client := utils.GetKubeClient()
+
+	err := client.CoreV1().Services(namespace).Delete(serviceName, &meta_v1.DeleteOptions{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
```
Comment (youngbupark) [inline pkg/kubernetes/kubernetes.go]: We would not need to check if err is nil. the below would be simpler.
```
return client.CoreV1().Services(namespace).Delete(serviceName, &meta_v1.DeleteOptions{})
```

**Comment by youngbupark** [comment]
@shalabhms Can we add unit-test for this ? I know we did not write any unit-test for these codes. at minimum, I would add unit-test for dapr_handler.

**Comment by shalabhms** [comment]
> @shalabhms Can we add unit-test for this ? I know we did not write any unit-test for these codes. at minimum, I would add unit-test for dapr_handler.

@youngbupark  and I discussed, need to first make kubernetes package testable and have tests around it.

**Comment by shalabhms** [comment]
> Seem like this needs unit tests?

Discussed with @youngbupark , need to first make kubernetes package testable and have unit tests. Agreed to merge this PR.

File: pkg/operator/handlers/dapr_handler_test.go [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+	"github.com/stretchr/testify/assert"
+
+	testclient "github.com/dapr/dapr/pkg/client/clientset/versioned"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestGetDaprID_WithValidId(t *testing.T) {
```
Comment (yaron2) [inline pkg/operator/handlers/dapr_handler_test.go]: Can you group tests per method with `t.Run()`?

See this reference:
https://github.com/dapr/dapr/blob/4c350c552066970be62cd1c15e155ffe38350e79/pkg/http/api_test.go#L58

**Pull request merged**